python_modules/libraries/dagster-components/dagster_components/core/schema/objects.py

```diff
@@ -11,26 +11,22 @@
 )
 from dagster._core.definitions.definitions_class import Definitions
 from dagster._record import replace
+from pydantic import BaseModel
 
-from dagster_components.core.schema.base import ComponentSchemaBaseModel
+from dagster_components.core.schema.base import ResolvableModel
 from dagster_components.core.schema.metadata import ResolvableFieldInfo
 from dagster_components.core.schema.resolver import TemplatedValueResolver
 
 
-def _post_process_key(resolved: Optional[str]) -> Optional[AssetKey]:
-    return AssetKey.from_user_string(resolved) if resolved else None
-
-
-class OpSpecBaseModel(ComponentSchemaBaseModel):
+class OpSpecModel(ResolvableModel["OpSpecModel"]):
```

Comment:
is there anyway to avoid the self reference here? It is really worth it to type `resolve` as `T`?

Reply:
ah nvm I withdraw my comment here